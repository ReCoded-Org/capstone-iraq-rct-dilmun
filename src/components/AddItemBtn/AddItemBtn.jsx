import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { useTranslation } from 'react-i18next'
import { useHistory } from 'react-router-dom'
import { OpenModal } from '../../redux/index'

export default function AddItemBtn() {
  const dispatch = useDispatch()
  const { t } = useTranslation()

  const user = useSelector(state => state.authentication)
  const history = useHistory()
  const handleClick = () => {
    if (user.isLoggedIn) {
      history.push('/addItem')
    } else {
      dispatch(OpenModal(true))
    }
  }
  return (
    <button
      type="button"
      onClick={handleClick}
      className="bg-blue  text-md   text-white py-2 text-center  space-x-1 items-center   w-full shadow-md hover:shadow-none rounded-full transition duration-300 ease-in focus:outline-none hover:bg-darkBlue px-3  "
    >
      <span className=""> {t('home.addItem')} </span>
      <FontAwesomeIcon icon="plus" />
    </button>
  )
}
