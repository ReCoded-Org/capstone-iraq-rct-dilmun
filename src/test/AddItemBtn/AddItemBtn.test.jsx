import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import AddItemBtn from '../../components/AddItemBtn'
import store from '../../redux/store'

it('Test: Add Item Button ', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <AddItemBtn />
        </Router>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
