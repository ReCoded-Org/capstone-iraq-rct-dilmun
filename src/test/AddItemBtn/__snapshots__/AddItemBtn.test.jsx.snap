// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Add Item Button  1`] = `
<button
  className="bg-blue  text-md   text-white py-2 text-center  space-x-1 items-center   w-full shadow-md hover:shadow-none rounded-full transition duration-300 ease-in focus:outline-none hover:bg-darkBlue px-3  "
  onClick={[Function]}
  type="button"
>
  <span
    className=""
  >
     
    home.addItem
     
  </span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-plus fa-w-14 "
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
